controllers/apps/transformer_component_workload.go

```diff
@@ -644,12 +646,18 @@ func (r *componentWorkloadOps) leaveMember4ScaleIn() error {
 			return switchoverErr
 		}
 
-		if err2 := lorryCli.LeaveMember(r.reqCtx.Ctx); err2 != nil {
+		if err2 := lorryCli.LeaveMember(r.reqCtx.Ctx, nil); err2 != nil {
 			// For the purpose of upgrade compatibility, if the version of Lorry is 0.7 and
 			// the version of KB is upgraded to 0.8 or newer, lorry client will return an NotImplemented error,
 			// in this case, here just ignore it.
 			if err2 == lorry.NotImplemented {
 				r.reqCtx.Log.Info("lorry leave member api is not implemented")
+			} else if unableToConnect(err2) {
+				r.reqCtx.Log.Info(fmt.Sprintf("when leaving pod %s by lorry, can not connect lorry on pod %s, try to leave member by other pods", pod.Name, pod.Name))
+				err3 := r.leaveMemberByOtherPods(desiredPods, pod)
+				if err == nil {
+					err = err3
+				}
```

Comment:
re-org `lorryCli.LeaveMember`, `unableToConnect` and `r.leaveMemberByOtherPods` into one single method of `r` seems more readable.